### PR TITLE
test(bigtable): unit test for AsyncRetryBulkApply

### PR DESCRIPTION
**Pull request by manish-qlogic**
This PR will close task#2 for #4100

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5288)

### CODE REVIEW

**Review by coryan — CHANGES_REQUESTED** [review]
Mostly cosmetic changes below.

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5288#-:-MK5mgnQEzJk_ayJ8UIu:bx675ls)* status: all files reviewed, 4 unresolved discussions (waiting on @manish-qlogic)

___
*[google/cloud/bigtable/internal/async_bulk_apply_test.cc, line 36 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5288#-MK5ldauEVdv9MQQUaVx:-MK5ldavDc-BWPQhMDRv:bv31gbg) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/28233a507573f5a0705b6046db853fb40c952164/google/cloud/bigtable/internal/async_bulk_apply_test.cc#L36)):*
> ```cpp
> 
> using ::google::cloud::testing_util::chrono_literals::operator"" _ms;
> using bigtable::testing::MockClientAsyncReaderInterface;
> ```

Please using `::` consistently.
___
*[google/cloud/bigtable/internal/async_bulk_apply_test.cc, line 74 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5288#-MK5luxVD8O6_MkwjFKm:-MK5luxVD8O6_MkwjFKn:b-7rf033) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/28233a507573f5a0705b6046db853fb40c952164/google/cloud/bigtable/internal/async_bulk_apply_test.cc#L74)):*
> ```cpp
>   auto* reader =
>       new MockClientAsyncReaderInterface<btproto::MutateRowsResponse>;
>   EXPECT_CALL(*reader, Read(_, _))
> ```

Here and below, you can just say `EXPECT_CALL(*reader, Read)`, no need to use the `_` placeholders (I know we use them everywhere else, we have stopped, but not cleaned up).
___
*[google/cloud/bigtable/internal/async_bulk_apply_test.cc, line 131 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5288#-MK5mVHUANuPeMTcd4-K:-MK5mVHUANuPeMTcd4-L:bc6f853) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/28233a507573f5a0705b6046db853fb40c952164/google/cloud/bigtable/internal/async_bulk_apply_test.cc#L131)):*
> ```cpp
>   };
> 
>   auto* reader =
> ```

nit: can you rename this `reader0`?
___
*[google/cloud/bigtable/internal/async_bulk_apply_test.cc, line 216 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5288#-MK5mZnnAYertLYFfTYW:-MK5mZnnAYertLYFfTYX:bjbbddy) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/28233a507573f5a0705b6046db853fb40c952164/google/cloud/bigtable/internal/async_bulk_apply_test.cc#L216)):*
> ```cpp
>   };
> 
>   auto* reader =
> ```

nit: rename this `reader0`?

**Review by manish-qlogic — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5288)* status: 2 of 3 files reviewed, 2 unresolved discussions (waiting on @coryan)

___
*[google/cloud/bigtable/internal/async_bulk_apply_test.cc, line 36 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5288#-MK5ldauEVdv9MQQUaVx:-MK7znVb0hpJe6E83PGj:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/28233a507573f5a0705b6046db853fb40c952164/google/cloud/bigtable/internal/async_bulk_apply_test.cc#L36)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Please using `::` consistently.
</blockquote></details>

Done.
___
*[google/cloud/bigtable/internal/async_bulk_apply_test.cc, line 74 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5288#-MK5luxVD8O6_MkwjFKm:-MK7zp9aEZmu-aZGLRyd:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/28233a507573f5a0705b6046db853fb40c952164/google/cloud/bigtable/internal/async_bulk_apply_test.cc#L74)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Here and below, you can just say `EXPECT_CALL(*reader, Read)`, no need to use the `_` placeholders (I know we use them everywhere else, we have stopped, but not cleaned up).
</blockquote></details>

Done.

**Review by coryan — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5288#-:-MK9eyLiEB8BtKqSEzjj:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

File: google/cloud/bigtable/internal/async_bulk_apply_test.cc [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        client_(new testing::MockDataClient) {}
+
+ protected:
+  std::shared_ptr<RPCRetryPolicy const> rpc_retry_policy_;
+  std::shared_ptr<RPCBackoffPolicy const> rpc_backoff_policy_;
+  std::shared_ptr<IdempotentMutationPolicy> idempotent_mutation_policy_;
+  MetadataUpdatePolicy metadata_update_policy_;
+  std::shared_ptr<FakeCompletionQueueImpl> cq_impl_;
+  CompletionQueue cq_;
+  std::shared_ptr<testing::MockDataClient> client_;
```
Comment (devbww) [inline google/cloud/bigtable/internal/async_bulk_apply_test.cc:60]: clang-diagnostic-shadow-field says:
```
non-static data member 'client_' of 'AsyncBulkApplyTest' shadows member inherited from type 'TableTestFixture'
```

**Pull request merged**